### ISSUE TITLE
Mobile testing checklist

### ISSUE THREAD

**Issue by dfarough** [wet-boew/wet-boew#552]
@wet-boew/access-working-group
I saw a good blog topic on this and thought it would be helpful when testing WET.

I am copying it here. the url is:
http://www.iheni.com/mobile-accessibility-tests/

Inspired by Al Duggin’s browser based tests for accessibility in his kick ass post building a web page with accessibility and interoperability in mind, I thought I’d put some tests together for mobile. This is intended as a guide you can use in day-to-day testing – you should be able to answer ‘yes’ to each question.

Tests should be carried out in the native browser in iOS Android or any other of your supported devices without any accessibility settings or support running as well as a combination of settings and support running i.e. zoom, voice output, inverse colours etc.

It’s worth also checking content in Firefox and Chrome on iOS and Android which both have voice output support and various zoom functions. I find this especially useful when I need to establish whether the content, voice output or browser is buggy.

Finally this is a general list of core tests. Different devices will offer different accessibility settings and support ranging from anything including external braille displays and keyboards to different types of zoom. It’s important to be aware of these and include them in your testing where possible.

Zoom and voice output disabled:
Is there sufficient contrast?
Does colour reinforce meaning rather than convey meaning alone?
Are links visually evident?
Are navigation cues clear?
Is pinch zoom supported (HTML only)?
Is the correct keyboard/input type used in forms i.e. tel, date, numbers, letters etc?
Can you complete all actions?
Do pop ups fit within the viewport i.e. you don’t have to swipe to find the close/submit/cancel buttons?
Do pop ups have a close button?
Is all content and functionality available by touch?
There is a clear visible focus on links, form fields, buttons etc when tested with a keyboard (Android)?
Text is selectable i.e. users can copy and paste and use speak aloud options?
Voice output enabled
Use the Web Rotor in iOS to test content on selections such as images, headings, landmarks, forms, hints, links, buttons etc. Try not to look at the screen when doing this or activate Screen Curtain if using iOS (3 finger triple tap).

Are images labelled appropriately?
Are decorative images ignored?
Is the content order logical?
Are landmarks labeled or have the appropriate heading announced with them?
Does the content order logical?
Do form fields have clear labels?
Is the appropriate keyboard used in forms i.e. tel, date, numbers, letters etc?
Are data table headings read correctly?
Are hints appropriate?
Users are notified of navigation cues i.e. if you can scroll pages/screen by swiping right when in portrait?
Are changes of state announced?
Does link text describe the target?
Are images and links to the same target grouped into one touchzone?
Can you complete all actions?
Is content in a different language read correctly?
Are buttons used for actions?
Have the correct controls (HTML) and traits (app) been used?
Is hidden content appropriate and necessary?
Do pop ups have a close button?
Does focus stay in the pop up rather than continue though the rest of the page/screen?
Is all content and functionality available by swiping left and right and up and down?
Zoom enabled
Text is readable?
Large areas of empty space are not present?
Labels and form inputs are not separated by large areas of empty space?
Voice output and zoom enabled
iOS 6 now allows both zoom and Voiceover to run simultaneously. Follow the same tests used with just zoom enabled as well as voice output running as well as:

Does visible text match audible labels on buttons, forms, linked images of text etc?
Does visible and Voiceover focus match?
Inverse colours
Follow the same tests used when zoom and voice output is disabled paying special attention to the following:

Is text readable?
Is there sufficient contrast?
Are navigation cues clear?
Ok, what have I missed? If you think of something let me know.

Tags: Android, Chrome, Firefox, iOS, MobA11y, Testing

Posted in Accessibility, Mobile

**Comment by hsrudnicki** [comment]
https://github.com/wet-boew/wet-boew/issues/5264 - New Homepage for WET Mockup

**Comment by hsrudnicki** [comment]
@thomasgohard and others, are you ready to close this issue or otherwise?

**Comment by hsrudnicki** [comment]
again, please update this issue status?

**Comment by hsrudnicki** [comment]
Please update this issue's status

**Comment by hsrudnicki** [comment]
Made "Going Stale" in order to close this issue (if no further response/updates) or otherwise and waiting for your comment(s)/update(s)?

**Comment by hsrudnicki** [comment]
@dfarough and others: Based on the last comment from @thomasgohard  of May 2013, this issue should have been resolved/fixed, please confirm/update this issue's status, whether this issue can be closed as resolved/fixed or if otherwise?
Thank you

**Comment by bsouster** [comment]
To do:
* [ ] Create a checklist in the documentation
